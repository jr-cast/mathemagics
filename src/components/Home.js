import React from 'react';
import './home.css';
import Navbar from './Navbar';

function Home() {
  return (
    <div>
      <div className="headerCont">
        <h1>Mathemagics</h1>
        <Navbar />
      </div>
      <h2>Welcome to our page!</h2>
      <p>
        Lorem ipsum dolor sit, amet consectetur adipisicing elit. Magnam id officiis,
        saepe asperiores itaque et corrupti eveniet iste atque, doloremque minus neque
        delectus ullam, error numquam voluptate animi enim nisi!
        Lorem ipsum dolor sit, amet consectetur adipisicing elit. Magnam id officiis,
        saepe asperiores itaque et corrupti eveniet iste atque, doloremque minus neque
        delectus ullam, error numquam voluptate animi enim nisi!
        Lorem ipsum dolor sit, amet consectetur adipisicing elit. Magnam id officiis,
        saepe asperiores itaque et corrupti eveniet iste atque, doloremque minus neque
        delectus ullam, error numquam voluptate animi enim nisi!
        Lorem ipsum dolor sit, amet consectetur adipisicing elit. Magnam id officiis,
        saepe asperiores itaque et corrupti eveniet iste atque, doloremque minus neque
        delectus ullam, error numquam voluptate animi enim nisi!
      </p>
      <p>
        Lorem ipsum dolor sit, amet consectetur adipisicing elit. Magnam id officiis,
        saepe asperiores itaque et corrupti eveniet iste atque, doloremque minus neque
        delectus ullam, error numquam voluptate animi enim nisi!
        Lorem ipsum dolor sit, amet consectetur adipisicing elit. Magnam id officiis,
        saepe asperiores itaque et corrupti eveniet iste atque, doloremque minus neque
        delectus ullam, error numquam voluptate animi enim nisi!
        Lorem ipsum dolor sit, amet consectetur adipisicing elit. Magnam id officiis,
        saepe asperiores itaque et corrupti eveniet iste atque, doloremque minus neque
        delectus ullam, error numquam voluptate animi enim nisi!
        Lorem ipsum dolor sit, amet consectetur adipisicing elit. Magnam id officiis,
        saepe asperiores itaque et corrupti eveniet iste atque, doloremque minus neque
        delectus ullam, error numquam voluptate animi enim nisi!
      </p>
    </div>
  );
}

export default Home;
