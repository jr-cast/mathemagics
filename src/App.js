import React from 'react';
import Home from './components/Home';

// eslint-disable-next-line react/prefer-stateless-function
class App extends React.Component {
  render() {
    return (
      <div className="container">
        <Home />
      </div>
    );
  }
}

export default App;
