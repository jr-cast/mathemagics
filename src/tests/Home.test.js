import { fireEvent, render, screen } from "@testing-library/react";
import React from "react";
import renderer from 'react-test-renderer';
import App from '../App';
import Home from '../components/Home';
import Navbar from '../components/Navbar';
import {
  BrowserRouter,
  MemoryRouter
} from 'react-router-dom';

describe('Testing Home Component', () => {

  test('renders all Home subelements correctly', () => {
    const page = renderer.create(<BrowserRouter><App><Home><Navbar /></Home></App></BrowserRouter>).toJSON();
    expect(page).toMatchSnapshot();
  });

  test('render home page first as index.html file', () => {
    render(<MemoryRouter><App /></MemoryRouter>);
    expect(screen.queryByText('Welcome to our page!')).not.toBeNull();
  });
})
