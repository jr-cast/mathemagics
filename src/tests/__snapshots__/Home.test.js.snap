// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Home Component renders all Home subelements correctly 1`] = `
<div
  className="container"
>
  <div>
    <div
      className="headerCont"
    >
      <h1>
        Mathemagics
      </h1>
      <nav
        className="navBar"
      >
        <ul>
          <li
            className="nav-link"
          >
            <a
              activeclassname="active-link"
              aria-current="page"
              className="active"
              exact="true"
              href="/"
              onClick={[Function]}
            >
              Home
            </a>
          </li>
          <li
            className="nav-link"
          >
            <a
              activeclassname="active-link"
              className=""
              exact="true"
              href="/calculator"
              onClick={[Function]}
            >
              Calculator
            </a>
          </li>
          <li
            className="nav-link"
          >
            <a
              activeclassname="active-link"
              className=""
              exact="true"
              href="/quote"
              onClick={[Function]}
            >
              Quote
            </a>
          </li>
        </ul>
      </nav>
    </div>
    <h2>
      Welcome to our page!
    </h2>
    <p>
      Lorem ipsum dolor sit, amet consectetur adipisicing elit. Magnam id officiis, saepe asperiores itaque et corrupti eveniet iste atque, doloremque minus neque delectus ullam, error numquam voluptate animi enim nisi! Lorem ipsum dolor sit, amet consectetur adipisicing elit. Magnam id officiis, saepe asperiores itaque et corrupti eveniet iste atque, doloremque minus neque delectus ullam, error numquam voluptate animi enim nisi! Lorem ipsum dolor sit, amet consectetur adipisicing elit. Magnam id officiis, saepe asperiores itaque et corrupti eveniet iste atque, doloremque minus neque delectus ullam, error numquam voluptate animi enim nisi! Lorem ipsum dolor sit, amet consectetur adipisicing elit. Magnam id officiis, saepe asperiores itaque et corrupti eveniet iste atque, doloremque minus neque delectus ullam, error numquam voluptate animi enim nisi!
    </p>
    <p>
      Lorem ipsum dolor sit, amet consectetur adipisicing elit. Magnam id officiis, saepe asperiores itaque et corrupti eveniet iste atque, doloremque minus neque delectus ullam, error numquam voluptate animi enim nisi! Lorem ipsum dolor sit, amet consectetur adipisicing elit. Magnam id officiis, saepe asperiores itaque et corrupti eveniet iste atque, doloremque minus neque delectus ullam, error numquam voluptate animi enim nisi! Lorem ipsum dolor sit, amet consectetur adipisicing elit. Magnam id officiis, saepe asperiores itaque et corrupti eveniet iste atque, doloremque minus neque delectus ullam, error numquam voluptate animi enim nisi! Lorem ipsum dolor sit, amet consectetur adipisicing elit. Magnam id officiis, saepe asperiores itaque et corrupti eveniet iste atque, doloremque minus neque delectus ullam, error numquam voluptate animi enim nisi!
    </p>
  </div>
</div>
`;
